### PR TITLE
Create example for useState

### PR DESCRIPTION
**Pull request by thanhtrantruong47**
Create example useState

Update state

https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

image

Updating objects and arrays in state 

https://react.dev/reference/react/useState#updating-objects-and-arrays-in-state

image

Resetting state with a key 

https://react.dev/reference/react/useState#resetting-state-with-a-key

image

### CODE REVIEW

File: examples/first-component/src/App.tsx [hunk 1]
```diff
@@ -8,6 +8,9 @@ import SyncedInputs from "./components/SyncedInput";
 import DrinkList from "./components/DrinkList";
 import PackingList from "./components/Item";
 import PackingListChallenge from "./components/ItemChallenge";
+import CounterState from "./components/useState/add-state";
```
Comment (thanhnguyen-agilityio) [inline examples/first-component/src/App.tsx]: Pls apply naming convention.

File: examples/first-component/src/components/Form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        image: e.target.value,
+      },
+    });
+  };
+
+  return (
+    <>
+      <label>
+        Name:
+        <input value={person.name} onChange={handleNameChange} />
```
Comment (thanhnguyen-agilityio) [inline examples/first-component/src/components/Form/index.tsx]: use destructuring

File: examples/first-component/src/App.tsx (hunk 1 above)
Comment (thanhtrantruong47) [reply examples/first-component/src/App.tsx]: Fixed

File: examples/first-component/src/components/Form/index.tsx (hunk 2 above)
Comment (thanhtrantruong47) [reply examples/first-component/src/components/Form/index.tsx]: Fixed

**Review by thanhnguyen-agilityio — APPROVED** [review]

**Pull request merged**